XCUITests/BookmarkingTests.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  BookmarkingTests.swift
+//  Client
+//
+//  Created by mozilla on 1/26/17.
+//  Copyright © 2017 Mozilla. All rights reserved.
+//
+
+import XCTest
+
+class BookmarkingTests: BaseTestCase {
+	var navigator: Navigator!
+	var app: XCUIApplication!
+	
+    override func setUp() {
+        super.setUp()
+        app = XCUIApplication()
+		navigator = createScreenGraph(app).navigator(self)
+	}
+	
+    override func tearDown() {
+      	super.tearDown()
+    }
+	
+	private func bookmark() {
+		navigator.goto(BrowserTabMenu)
+		app.collectionViews.cells["AddBookmarkMenuItem"].tap()
+		navigator.nowAt(BrowserTab)
+	}
+	
+	private func unbookmark() {
+		navigator.goto(BrowserTabMenu)
+		app.cells["Remove Bookmark"].tap()
+		navigator.nowAt(BrowserTab)
+	}
+	
+	private func checkBookmarked() {
+		navigator.goto(BrowserTabMenu)
+		waitforExistence(app.collectionViews.cells["RemoveBookmarkMenuItem"])
+		navigator.goto(BrowserTab)
+	}
+	
+	private func checkUnbookmarked() {
+		navigator.goto(BrowserTabMenu)
+		waitforExistence(app.collectionViews.cells["AddBookmarkMenuItem"])
+		navigator.goto(BrowserTab)
+	}
+	
+    func testBookmarkingUI() {
+		let url1 = "www.google.com"
+		let url2 = "www.mozilla.org"
+		
+		// Go to a webpage, and add to bookmarks, check it's added
+		navigator.openNewURL(url1)
+		bookmark()
+		checkBookmarked()
+		
+		// Load a different page on a new tab, check it's not bookmarked
+		//navigator.goto(NewTabScreen)
+		navigator.openNewURL(url2)
+		checkUnbookmarked()
+		
+		// Go back, check it's still bookmarked, check it's on bookmarks home panel
+		navigator.goto(TabTray)
+		app.collectionViews.cells["Google"].tap()
+		navigator.nowAt(BrowserTab)
+		checkBookmarked()
+		
+		// Open it, then unbookmark it, and check it's no longer on bookmarks home panel
+		unbookmark()
+		checkUnbookmarked()
+		
+    }
+    
+}
```

Comment:
Nits: whitespace.

Indent is four spaces. I think you have some tabs mixed in there too.